compiler/rustc_session/src/errors.rs

```diff
@@ -324,7 +341,8 @@ pub fn report_lit_error(sess: &ParseSess, err: LitError, lit: token::Lit, span:
             if looks_like_width_suffix(&['i', 'u'], &suf) {
                 // If it looks like a width, try to be helpful.
                 sess.emit_err(InvalidIntLiteralWidth { span, width: suf[1..].into() });
-            } else if let Some(fixed) = fix_base_capitalisation(suf) {
+            } else if let Some(fixed) = fix_base_capitalisation(suf)
+                && looks_like_base_prefixed_number(symbol.as_str(), fix_base_capitalisation(suf)) {
```

Comment:
We only want to make this suggestion if the literal starts with `0X`, `0O`, or `0B`. So I *think* this can be as simple as this:
```
            } else if symbol == sym::integer(0) && let Some(fixed) = fix_base_capitalisation(suf) {
```
and `looks_like_base_prefixed_number` isn't necessary.